### ISSUE TITLE
Add an option to edit a JSON or Vega JSON file being displayed

### ISSUE THREAD

**Issue by JasonWeill** [jupyterlab/jupyterlab#14280]
### Problem

Found using 4.0.0b0.

When I open a `.json` file, the default UX is a non-interactive viewer:

![image](https://user-images.githubusercontent.com/93281816/228002983-4a610092-850f-4d2c-a31b-1ef35c9537fb.png)

… and for a `.vg.json` file, the default UX is a Vega render:

![image](https://user-images.githubusercontent.com/93281816/228003050-3f9b0ddf-709f-45d3-b419-a2a2c150220f.png)

If I right-click on the file in the file browser, I can use the "Open With" menu to manually choose the editor:

![image](https://user-images.githubusercontent.com/93281816/228003170-ea34f190-dccf-4af5-bf8b-fc0ec74e1d7c.png)

… but when I'm viewing a file in a read-only mode as shown in the first two screen shots, I couldn't find a way to edit what I'm seeing. The Vega render has an "Open in Vega Editor" mode, but this redirects me to https://vega.github.io/editor/ with the JSON code present, and takes me out of JupyterLab.

![image](https://user-images.githubusercontent.com/93281816/228003382-a87d9d57-5b17-44df-b666-e9be9fc4a59a.png)

### Proposed Solution

When I'm in a read-only mode for JSON or Vega JSON, add a prominent button to switch into "Edit" mode.

When I'm in "Edit" mode, add a prominent button to switch into "View" or "Render" mode.

### Additional context

See also #4168 for bad behavior when attempting to render malformed Vega JSON.

**Comment by ellisonbg** [comment]
If you right click the file in the file browser, you can open with the text editor to edit the JSON. This was by design as it enables you to open both the rendered version and text file at the same time (in JupyterLab) and as you edit the text file the rendered version will automatically update. This works for any file type in JupyterLab that is 1) text based and 2) has multiple output formats.

**Comment by JasonWeill** [comment]
I agree, and that's how I worked around it; I'd prefer if the viewer UI had a button to launch an editor, without using the file browser.
